### PR TITLE
Add android-sdk salt state

### PR DESCRIPTION
**Pull request by wbrawner**

### CODE REVIEW

File: salt/android-sdk/init.sls [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+android-sdk:
+    file.managed:
+        - name: /usr/local/src/sdk-tools-linux-{{ version }}.zip
+        - source: https://dl.google.com/android/repository/sdk-tools-linux-{{ version  }}.zip
+        - source_hash: {{ source_hash }}
+
+    cmd.watch:
+        - name: unzip -d /opt/android-sdk /usr/local/src/sdk-tools-linux-{{ version }}.zip &&
+                find android-sdk -type f -exec chmod 644 {} \; &&
```
Comment (thusoy) [inline salt/android-sdk/init.sls]: The paths given to find here doesn't seem to align with where stuff is unpacked. Is the permission modification necessary here at all?

**Comment by thusoy** [comment]
You might also want to include a link to where the version spec can be found somewhere, either in the state or the readme: `https://developer.android.com/studio/index.html#command-tools`

File: salt/android-sdk/init.sls (hunk 1 above)
Comment (thusoy) [reply salt/android-sdk/init.sls]: Actually, this probably breaks the sdk as it removes the executable bit from the binaries. But I see that as it's shipped the binaries are only executable by root, where the intention is probably that they should be executed by everyone. Maybe find existing executable binaries and make them 755 too?

**Comment by wbrawner** [comment]
One more thing on this, since root owns the `/opt/android-sdk` directory, only root can make the calls to install/update new tools. Should we create an `android` group that has read/write access to these files or do we leave it up to the implementer to change the ownership of the files?

**Comment by thusoy** [comment]
An `android` group sounds good to me.

File: salt/android-sdk/init.sls [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+android-sdk:
+    file.managed:
+        - name: /usr/local/src/sdk-tools-linux-{{ version }}.zip
+        - source: https://dl.google.com/android/repository/sdk-tools-linux-{{ version  }}.zip
+        - source_hash: {{ source_hash }}
+
+    cmd.watch:
+        - name: unzip -d /opt/android-sdk /usr/local/src/sdk-tools-linux-{{ version }}.zip &&
+                find /opt/android-sdk -type f -exec test -x {} \; -exec chmod 755 {} \; &&
```
Comment (thusoy) [inline salt/android-sdk/init.sls]: `find` takes a `-perm` argument, wouldn't that make more sense than forking out to `test`?

File: salt/android-sdk/init.sls [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{% set version_id = '3859397 sha256=444e22ce8ca0f67353bda4b85175ed3731cae3ffa695ca18119cbacef1c1bea0' %}
+{% set version, source_hash = version_id.split() %}
+
+android-sdk-deps:
+    pkg.installed:
+        - name: unzip
+
+android-sdk:
+    group.present:
+        - name: android
```
Comment (thusoy) [inline salt/android-sdk/init.sls]: Add a newline after this and it's golden.

**Comment by wbrawner** [comment]
Merged #9

**Pull request merged**